sdk/test/logs/batch_log_processor_test.cc

```diff
@@ -176,18 +187,24 @@ TEST_F(BatchLogProcessorTest, TestAsyncShutdown)
   const std::chrono::milliseconds export_delay(0);
   const std::chrono::milliseconds scheduled_delay_millis(5000);
   const size_t max_export_batch_size = 512;
-  const size_t max_queue_size        = 2048;
+  const size_t max_queue_size        = 20;
   const bool is_export_async         = true;
+  const size_t max_export_async      = 5;
 
   auto batch_processor = GetMockProcessor(logs_received, is_shutdown, is_export_completed,
                                           export_delay, scheduled_delay_millis, max_queue_size,
-                                          max_export_batch_size, is_export_async);
+                                          max_export_batch_size, is_export_async, max_export_async);
 
   // Create a few test log records and send them to the processor
-  const int num_logs = 3;
+  const int num_logs = 500;
 
   for (int i = 0; i < num_logs; ++i)
   {
+    if (i % 20 == 0)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
Why do we need sleep here?I think it's better to use a condition_variable to get notify if we want to know when jobs are done and it's ready to send next round datas.
Sometimes the CI runner is slow and can not finish this job in 1 millisecond.

Reply:
ok will take care this.

Reply:
I mean `while (CleanUpGarbageAsyncData() == false);` is a busy wait. `CleanUpGarbageAsyncData` only lock `async_export_data_m`, and this lock has a very small critial section, it's almost busy wait when worker thread is exited when exporter is still exporting data in other thread.